test/integration/integration_test.rb

```diff
@@ -83,6 +85,7 @@ def ipv6
 
   def purge_cache name
     %x{#{ENV['PDNS_CONTROL'] || "pdns_control"} purge "#{name}"}
-    $? == 0
+    sleep 60 unless $? == 0
```

Comment:
Wow, 1 minute wait -- that might make this a very long-running test? Perhaps reducing ttl to an absurdly small value would help?